### PR TITLE
Various fixes to make testing pass in async-branch

### PR DESCRIPTION
**Pull request by bouwew**

### CODE REVIEW

**Comment by bouwew** [comment]
There is still an error in testing:
```
--- Logging error ---
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/logging/__init__.py", line 1163, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
Call stack:
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/base_events.py", line 1870, in call_exception_handler
    ctx.run(self._exception_handler, self, context)
  File "/home/runner/work/plugwise_usb-beta/plugwise_usb-beta/ha-core/tests/conftest.py", line 584, in exc_handle
    orig_exception_handler(loop, context)
  File "/home/runner/work/plugwise_usb-beta/plugwise_usb-beta/ha-core/homeassistant/runner.py", line 103, in _async_loop_exception_handler
    logger.error(
Message: 'Error doing job: %s (%s): %s'
```

But apparently not blocking.

File: scripts/setup.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ if [ -f "${my_venv}/bin/activate" ]; then
     # shellcheck disable=SC1091
     . "${my_venv}/bin/activate"
     # Install commit requirements
+    python3 -m ensurepip --upgrade
```
Comment (ArnoutD) [inline scripts/setup.sh:14]: you can better use --seed in the venv creation
https://docs.astral.sh/uv/reference/cli/#uv-venv

**Review by ArnoutD — APPROVED** [review]
most imporant test and commits continue .... i'll put my remark as PR on top

File: scripts/setup.sh (hunk 1 above)
Comment (CoMPaTech) [reply scripts/setup.sh:14]: Good call, goes for the Smile stuff too - we might need to re-visit all of them

**Pull request merged**